--- origsrc/aspell-0.60.5/Makefile.am	2006-12-10 21:36:41.000000000 -0600
+++ src/aspell-0.60.5/Makefile.am	2007-12-11 09:40:26.908890500 -0600
@@ -129,6 +129,8 @@
 
 aspell_SOURCES = prog/aspell.cpp prog/check_funs.cpp prog/checker_string.cpp
 
+aspell_CXXFLAGS = $(CURSES_INCLUDE)
+
 aspell_LDADD = libaspell.la $(CURSES_LIB)
 
 prezip_bin_SOURCES = prog/prezip.c
@@ -194,7 +196,7 @@
 else # not COMPILE_IN_FILTERS
 
 dynamic_optfiles += ${optfiles}
-filter_ldflags = -module -avoid-version
+filter_ldflags = -module -avoid-version -no-undefined
 
 ### Add name of filter library containing your filter. Name always
 ### must look like lib<filtername>-filter.la see development manual
--- origsrc/aspell-0.60.5/lib/new_filter.cpp	2005-06-19 19:15:15.000000000 -0500
+++ src/aspell-0.60.5/lib/new_filter.cpp	2007-12-11 09:01:08.272048500 -0600
@@ -456,13 +456,23 @@
     assert(slash);
     if (module->file.empty()) {
       module->file.assign(option_file.str(), slash + 1 - option_file.str());
-      //module->file += "lib";
+#ifdef __CYGWIN__
+      module->file += "cyg";
+      module->file += filter_name;
+      module->file += "-filter.dll";
+#else
+      module->file += "lib";
       module->file += filter_name;
       module->file += "-filter.so";
+#endif
     } else {
       if (module->file[0] != '/')
         module->file.insert(0, option_file.str(), slash + 1 - option_file.str());
+#ifdef __CYGWIN__
+      module->file += ".dll";
+#else
       module->file += ".so";
+#endif
     }
 
     return module.release();
